File: JavaChatSockets/ChatPersistence/src/main/java/chat/persistence/repository/mock/UserRepositoryMock.java
package chat.persistence.repository.mock;

import chat.model.User;
import chat.persistence.RepositoryException;
import chat.persistence.UserRepository;
import org.springframework.stereotype.Component;

import java.util.*;


@Component
public class UserRepositoryMock implements UserRepository{
    private Map<String, User> allUsers;

    public UserRepositoryMock() {
        allUsers=new TreeMap<String, User>();
        populateUsers();
    }


    private static final Set<User> noFriends=new TreeSet<>();


    private void populateUsers(){
        User ana=new User("ana", "ana", "Popescu Ana");
        User mihai=new User("mihai", "mihai", "Ionescu Mihai");
        User ion=new User("ion", "ion", "Vasilescu Ion");
        User maria=new User("maria", "maria", "Marinescu Maria");
        User test=new User("test", "test", "Test user");

        ana.addFriend(mihai);
        ana.addFriend(test);

        mihai.addFriend(ana);
        mihai.addFriend(test);
        mihai.addFriend(ion);

        ion.addFriend(maria);
        ion.addFriend(test);
        ion.addFriend(mihai);

        maria.addFriend(ion);
        maria.addFriend(test);

        test.addFriend(ana);
        test.addFriend(mihai);
        test.addFriend(ion);
        test.addFriend(maria);

        allUsers.put(ana.getId(),ana);
        allUsers.put(mihai.getId(), mihai);
        allUsers.put(ion.getId(), ion);
        allUsers.put(maria.getId(), maria);
        allUsers.put(test.getId(), test);


    }

    @Override
    public User findBy(String username) {
        User userR=allUsers.get(username);
        if (userR==null)
            return null;
       // User res=new User(username);
       // res.setName(userR.getName());
        return userR;

    }

    @Override
    public User findBy(String username, String pass) {
        User userR=allUsers.get(username);
        if ((userR!=null)&&(userR.getPasswd().equals(pass)) ){
            User res = new User(username);
            res.setName(userR.getName());
            return res;
        }
        return null;

    }

    @Override
    public Iterable<User> getFriendsOf(User user) {
        User userR=allUsers.get(user.getId());
        if (userR!=null)
            return userR.getFriends();

        return noFriends;
    }

    @Override
    public User[] getUsers() {
        return allUsers.values().toArray(new User[allUsers.size()]);
    }

    @Override
    public void save(User user) {
        System.out.println("[UserRepositoryMock] save user - entering");
        if(allUsers.containsKey(user.getId()))
            throw new RepositoryException("Username already exists: "+user.getId());
        allUsers.put(user.getId(),user);
        System.out.println("[UserRepositoryMock] save user - exiting ok");
    }

    @Override
    public void delete(String s) {
        if (allUsers.containsKey(s))
            allUsers.remove(s);
        else
            throw new RepositoryException("Username ["+s+"] not found for deletion.");
    }

    @Override
    public User findOne(String s) {

        return findBy(s);
    }

    @Override
    public void update(String s, User user) {
        if (allUsers.containsKey(s)){
            if (s.equals(user.getId())) {
                allUsers.put(s, user);
                return;
            }

        }
         throw new RepositoryException("User could not be updated "+user);
    }

    @Override
    public Iterable<User> getAll() {
        return allUsers.values();
    }
}

File: ChatSockets/persistence/UserRepositoryMock.cs
using System;
using System.Collections.Generic;
using chat.model;

namespace chat.persistence
{
    namespace repository.mock
    {
        public class UserRepositoryMock: IUserRepository{
            private readonly IDictionary<String, User> allUsers;

            public UserRepositoryMock() {
                allUsers=new Dictionary<String, User>();
                populateUsers();
            }
            public User findBy(string username, string pass)
            {
                User userR;
                bool  ok=allUsers.TryGetValue(username,out userR);
                if (!ok || userR.Password!=pass)
                    return null;
                return new User(username, "",userR.Name);
            }

            public IEnumerable<User> getFriendsOf(User user)
            {
                User userR;
                if (allUsers.TryGetValue(user.Id, out userR))
                    return userR.Friends;
                return new User[0];
            }
            private void populateUsers(){
                User ana=new User("ana", "ana", "Popescu Ana");
                User mihai=new User("mihai", "mihai", "Ionescu Mihai");
                User ion=new User("ion", "ion", "Vasilescu Ion");
                User maria=new User("maria", "maria", "Marinescu Maria");
                User test=new User("test", "test", "Test user");

                ana.addFriend(mihai);
                ana.addFriend(test);

                mihai.addFriend(ana);
                mihai.addFriend(test);
                mihai.addFriend(ion);

                ion.addFriend(maria);
                ion.addFriend(test);
                ion.addFriend(mihai);

                maria.addFriend(ion);
                maria.addFriend(test);

                test.addFriend(ana);
                test.addFriend(mihai);
                test.addFriend(ion);
                test.addFriend(maria);

                allUsers[ana.Id]=ana;
                allUsers[mihai.Id]= mihai;
                allUsers[ion.Id]= ion;
                allUsers[maria.Id]= maria;
                allUsers[test.Id]= test;


            }


            public void save(User entity)
            {

            }

            public void delete(string id)
            {
                throw new NotImplementedException();
            }

            public User findOne(string id)
            {
                throw new NotImplementedException();
            }

            public void update(string id, User e)
            {
                throw new NotImplementedException();
            }

            public IEnumerable<User> getAll()
            {
                throw new NotImplementedException();
            }

            public User findBy(string username)
            {
                throw new NotImplementedException();
            }


        }
    }
}